warehouse/forklift/metadata.py

```diff
@@ -233,6 +233,19 @@ def _validate_metadata(metadata: Metadata, *, backfill: bool = False):
                         )
                     )
 
+    # Ensure that License and License-Expression are mutually exclusive
+    # See https://peps.python.org/pep-0639/#deprecate-license-field
+    if metadata.license and metadata.license_expression:
+        errors.append(
+            InvalidMetadata(
+                "license",
+                (
+                    "License is deprecated when License-Expression is present. "
+                    "Only License-Expression should be present."
+                ),
+            )
+        )
+
```

Comment:
Seems like this is a check that should be happening in pypa/packaging instead, but isn't?

```
>>> Metadata.from_raw({"metadata_version": "2.4", "license": "MIT", "license_expression": "MIT", "version": "6.6.6", "name": "throwaway"})
<packaging.metadata.Metadata object at 0x7f44b9647680>
```

Reply:
That validation isn't backed into packaging as the [PEP](https://peps.python.org/pep-0639/#deprecate-license-field) says "tools MAY issue a warning" and "PyPI MUST reject"

Reply:
I'm not sure if I agree or disagree here, mainly because I am not certain where the responsibility belongs.

This specific check does ensure that PyPI's implementation meets the specification that

> For all newly-uploaded [distribution archive](https://packaging.python.org/en/latest/glossary/#term-Distribution-Archive)s that include a License-Expression field, the [Python Package Index (PyPI)](https://pypi.org/) MUST reject any that specify both License and License-Expression fields.

No matter the underlying validation, so I'm prone to remain explicit and keep it.

Reply:
I think we can leave this for now and figure out later if it should belong upstream!